### PR TITLE
feat: Policy Output

### LINKED ISSUE #608: Policy Output Improvements
Short Term Implementation:
- [x] #609
- [x] #610
- [ ] #611

### LINKED ISSUE #609: Policy HCL Changes:
-  [x] For providers that need to have more context around the results of a policy result we will enable a policy to declare columns that all checks must return. For example required (ie project_id, region, account), 
- [x] Declare the field is the `PK/ID` (for future decorator)
-  [ ] All SQL queries should have the following fields outputs should have the following:
    - `CQ_ID` (suggested)
    - `Status` (suggested)
    - `Reason` (suggested)
    - `Remediation` (suggested)
-  [ ] For output on a specific  Can add:
    - `Reason`
    - `Remediation`

### LINKED ISSUE #610: Policy Validate- Create a CLI command that can ensure policy and checks are valid

### LINKED ISSUE #611: Policy CLI Output
Improve UI/Output —> more terminal less emoji

### PR DESCRIPTION
**Pull request by roneli**
Closes #608 

Closes #609 
Closes #610 
Closes #611

### CODE REVIEW

File: pkg/policy/execute.go [hunk 1]
```diff
@@ -65,12 +69,45 @@ type Executor struct {
...
+
+func (r Rows) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+
+func (r Rows) Less(i, j int) bool {
+	r1 := r[i]
+	r2 := r[j]
+	for l := 0; l < len(r1.Identifiers); l++ {
+		if cast.ToString(r1.Identifiers[l]) > cast.ToString(r2.Identifiers[l]) {
```
Comment (disq) [inline pkg/policy/execute.go:106]: Is this descending sort?

Comment (roneli) [reply pkg/policy/execute.go:106]: ill add a test to this, but does it matter? the sort is mostly to keep it consistent with same data

Comment (disq) [reply pkg/policy/execute.go:106]: test would be good yes

File: pkg/policy/validate.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+func validatePolicy(ctx context.Context, storage database.Storage, policy *Policy, identifiers []string, policyPath string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	if len(policy.Identifiers) > 0 {
+		log.Debug().Strs("previous_identifiers", identifiers).Strs("identifiers", policy.Identifiers).Msg("overriding parent policy identifiers")
+		identifiers = policy.Identifiers
+	}
+	if identifiers == nil {
+		diags = diags.Add(diag.FromError(fmt.Errorf("policy %s has no identifiers set", path.Join(policyPath, policy.Name)), diag.USER, diag.WithSeverity(diag.WARNING)))
```
Comment (disq) [inline pkg/policy/validate.go:44]:
```suggestion
		diags = diags.Add(diag.FromError(fmt.Errorf("policy %q has no identifiers set", path.Join(policyPath, policy.Name)), diag.USER, diag.WithSeverity(diag.WARNING)))
```

I suggest using `%q` when referring to user input.

File: pkg/policy/validate.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		columns, dd := getQueryColumns(ctx, conn, c.Name, c.Query)
+		if dd.HasErrors() {
+			return diags.Add(dd)
+		}
+		// Check identifiers exist
+		for _, id := range identifiers {
+			if funk.InStrings(columns, id) {
+				continue
+			}
+			diags = diags.Add(diag.FromError(fmt.Errorf("check %s is missing identifier %s", path.Join(policyPath, c.Name), id),
```
Comment (disq) [inline pkg/policy/validate.go:80]: same here with `%q`

File: pkg/policy/validate.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+				diag.USER, diag.WithSeverity(diag.WARNING), diag.WithDetails("Check that query returns the following identifier: %s", id)))
+		}
+		// Check for cq_meta columns
+		if c.Reason == "" {
+			if !funk.InStrings(columns, "cq_reason") {
+				diags = diags.Add(diag.FromError(fmt.Errorf("check %s doesn't define reason in configuration or query", path.Join(policyPath, c.Name)),
```
Comment (disq) [inline pkg/policy/validate.go:86]: and here

File: pkg/ui/console/client.go [hunk 5]
```diff
@@ -365,7 +365,7 @@ func (c Client) RunPolicies(ctx context.Context, policySource, outputDir string,
 		OutputDir:   outputDir,
 		RunCallback: policyRunCallback,
 	})
-	if resp != nil {
+	if resp == nil {
```
Comment (disq) [inline pkg/ui/console/client.go:368]: I think these were fixed and rebase will conflict.

Comment (roneli) [reply pkg/ui/console/client.go:368]: ill fix the conflict on the rebase.

**Pull request merged**